### PR TITLE
WT-13043 Introduce Ruff checks

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: dist/ruff.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+required-version = "==0.4.5"
+
+extend-exclude = [
+    "../test/3rdparty",
+]
+
+[lint]
+ignore = [
+    "E401", # multiple-imports-on-one-lines
```
Comment (ajmorton) [inline dist/ruff.toml:9]:
```suggestion
    "E401", # multiple-imports-on-one-line
```

File: dist/ruff.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    "E402", # module-import-not-at-top-of-file
+    "E701", # multiple-statements-on-one-line-colon
+    "E702", # multiple-statements-on-one-line-semicolon
```
Comment (ajmorton) [inline dist/ruff.toml:12]: Thanks for adding the error names! Much easier to work with than googling error numbers

File: dist/ruff_check.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    doc_link = "https://docs.astral.sh/ruff/installation/"
+    lines = []
+    ruff_version = None
+    with open(ruff_config) as file:
+        for line in file:
+            if m := re.search(r'required-version = "==(\d.\d.\d)"', line.strip()):
+                ruff_version = m.group(1)
+                break
+    print("Ruff is not installed!")
+    print("\nCreate a virtual environment if you haven't yet:")
```
Comment (ajmorton) [inline dist/ruff_check.py:37]: Is this the first time we're mandating a virtualenv in order to run `s_all`? I've never needed one in the past.

Comment (etienneptl) [reply dist/ruff_check.py:37]: I am happy to change the wording here. It's really optional, everyone is free to manage their local packages the way they want.

Comment (etienneptl) [reply dist/ruff_check.py:38]: @pmacko86 suggested that `s_all` would create its own env, which is a possible improvement that we can do in a different ticket.

Comment (ajmorton) [reply dist/ruff_check.py:38]: SGTM. I'd prefer not mentioned virtualenv in the comment at the moment and have a generic "Please install ruff"

**Review by ajmorton — APPROVED** [review]
LGTM. Thanks!

**Comment by ershov** [comment]
Can you automatically install ruff in a local directory if it's not present?
Giving the user instruction what to do is good but why not just do it automatically?

**Review by ershov — APPROVED** [review]
lgtm!

**Pull request merged**